this is the first project

Meskerem3 opened the pull request:
This project is the initial  project which is the base for the final project in this project I simply create the class for teacher,student and person with out the main class and based on the requirement I add inheritance and override and I think I fulfill the project requirement in some way.

Commit:
this is the first project

Benawi commented on Person.java:
```diff
@@ -0,0 +1,45 @@
+    private int id;
+    private String name;
+    private int age;
+    private boolean parent_Permission;
```
- Modifiers not requested now

Benawi commented on Person.java:
```diff
@@ -0,0 +1,45 @@
+    private int age;
+    private boolean parent_Permission;
+
+    public Person(int id, String name, int age, boolean parent_Permission) {
```
- default values of name and parent_permission ?

Benawi commented on Person.java:
```diff
@@ -0,0 +1,45 @@
+
+    public Person(int id, int age) {
+        this(id, "unknown", age, true);
+    }
```
- Not requested to create this constructor

Benawi commented on Student.java:
```diff
@@ -0,0 +1,16 @@
+public class Student extends Person {
+    private String classroom;
+
+    public String getClassroom() {
+        return classroom;
+    }
+
+    public void setClassroom(String classroom) {
+        this.classroom = classroom;
+    }
```
- Not requested

Benawi commented on Teacher.java:
```diff
@@ -0,0 +1,21 @@
+    public Teacher(int id, String name, int age, boolean parentPermission, String specialization) {
+        super(id, name, age, parentPermission);
+        this.specialization = specialization;
+    }
+    public String
+    getSpecialization(){
+        return specialization;
+    }
+    public void setSpecialization(String specialization){
+        this.specialization = specialization;
+    }
```
- setter and getter method not requested for this att

Benawi reviewed (changes requested):
#### Hi @Meskerem3 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

![giphy](https://user-images.githubusercontent.com/25227022/169391888-8dd8108f-0d42-4e54-87cc-a3f9fd392dbc.gif)


## Highlights

✔️  Create Person Class 
✔️  Create Student Class 
✔️  Create Teacher Class 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. @benawi

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Meskerem3 committed:
Update Person.java

I update the code as your suggestion

Meskerem3 committed:
Update Student.java

In this code i used the setter and getter method but i should not have to use it so i update the code as the instruction

Meskerem3 committed:
Update Teacher.java

In this class it was not allowed  to use the setter and getter method but i use it and now i update the code as your suggestiion
